code/game/objects/effects/particle_holder.dm

```diff
@@ -59,3 +59,46 @@
 /// See [https://www.byond.com/docs/ref/#/{notes}/particles] for position documentation
 /obj/effect/abstract/particle_holder/proc/set_particle_position(list/pos)
 	particles.position = pos
+
+///Stores an index of particles and their instances for reusing : list(particle_type = list(INSTANCES BABEEYYYY))
+GLOBAL_LIST_EMPTY(shared_particle_holders)
+
+///Give a particle effect to an atom (or turf the definition is a lie) from a global list. Use this if you have a lot of particle holders on someones
+///screen and client performance is a problem. This way it's only calculated x amount of times and you can replicate it without putting too much extra burden on clients
+/proc/get_shared_particle_effect(atom/atom, pool_size, particle_effect, particle_flags = NONE)
```

Comment:
What happens if the same effect is requested, but with a different pool size?
I wonder if a better pattern wouldn't be subtypes of /obj/effect/abstract/particle_holder_shared with flags/pool size/effect set, or at least a holder datum